### DISCUSSION TITLE
ログイン後、そのままアクセスしたいページにリダイレクトする設定

### DISCUSSION THREAD

**Discussion by newspot4034** [growilabs/growi#9029] [Ideas / Requests]
いつも大変お世話になっております。
職場でナレッジ共有のためのWikiをGrowiで運用しております。
利用者からいくつか改善要望が上がったものがあり、以下リクエストさせてもらえればと思います。

# 環境
- Growi 6.3.4
- アクセス履歴管理のため、ゲストアクセス不可の設定済 ※ログイン必須の環境

# 現在の動作
- 非ログインユーザーが、GrowiのページのURLを直接叩いた場合、ログイン画面にリダイレクトする
- ログイン認証後はトップページに遷移する

# 機能追加リクエスト
- ログイン時にもともとアクセスしているURLにリダイレクトするオプションの追加
- URLパラメータ等で元々のアクセス元のURLを保持し、ログイン後はそちらにリダイレクトさせる

# 追加要望の経緯
社内で動いている別システムにGrowiに掲載されている記事のURLを記載することがあり、
普段あまりGrowiを閲覧しないユーザだとログイン画面に遷移後、トップページに戻されてしまう状態となっている。
Growiに初めてアクセスするユーザがトップページに戻された結果、
該当URLのリンクが切れていると勘違いしてしまう事例があったため本リクエストを上げさせていただきました。

**Comment by yuki-takei** [comment]
当該要望の機能は既に実装されている認識で、今本番運用している v7.0.16 のシステムではログイン後にリダイレクトされることを確認しました。(こちらもゲストアクセス不可の環境です)

期待される動作にならない可能性としては以下が考えられます。

- v6.3.4 でその機能がデグレしている
- 期待される動作にならない条件が他にある

いずれにせよ問題の切り分けが必要かなと思います。
もし問題を再現できる条件が判明したら教えてください。

因みにリダイレクトの実装としてはセッションを利用しています。

1. https://github.com/weseek/growi/blob/v7.0.16/apps/app/src/server/middlewares/login-required.js#L59
1. https://github.com/weseek/growi/blob/v7.0.16/apps/app/src/server/routes/login.js#L97

**Reply by newspot4034** [reply]
お返事いただきありがとうございます。
会社で別のGrowi(7.0.13)で試したところ、同様にリダイレクト処理が走らないことを確認しております。

色々こちらで切り分けを行ったのですが、どうやら会社でユーザID/パスワードの統一化のために
LDAP認証を用いているのですがこちらが原因のように思われます。

# 検証結果
- 弊社環境で試した結果、LDAP認証を用いてアクセスした場合にリダイレクトが動作しないように見える
- ローカルユーザ認証だとご指摘の通り問題なくリダイレクト処理が働く

## v6.3.4
ローカルユーザ：リダイレクトあり
LDAP認証：リダイレクトされず、トップページに戻る

## v7.0.13
ローカルユーザ：リダイレクトあり
LDAP認証：リダイレクトされず、トップページに戻る
